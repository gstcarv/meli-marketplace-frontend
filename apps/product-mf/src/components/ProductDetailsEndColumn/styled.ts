import { breakpoints } from '@meli/design-tokens';
import styled from 'styled-components';

export const Container = styled.section`
    width: 100%;
    flex: 0.25;
    padding: ${({ theme }) => theme.spacing.get('lg')};

    @media screen and (max-width: ${breakpoints.get('lg')}) {
        padding: ${({ theme }) => theme.spacing.get('xs')};
    }
`;

export const InfoHeading = styled.div`
    display: flex;
`;

export const InfoHeadingText = styled.span`
    display: flex;
    color: ${({ theme }) => theme.colors.get('gray-600')};
    font-size: ${({ theme }) => theme.fontSize.get('caption')};
    font-weight: ${({ theme }) => theme.fontWeight.get('regular')};
    margin-bottom: ${({ theme }) => theme.spacing.get('xs')};
`;

export const ProductTitle = styled.h1`
    color: #000000;
    font-size: ${({ theme }) => theme.fontSize.get('h4')};
    font-weight: ${({ theme }) => theme.fontWeight.get('bold')};
    margin-bottom: 0;

    @media screen and (max-width: ${breakpoints.get('lg')}) {
        font-size: ${({ theme }) => theme.fontSize.get('h5')};
    }
`;

export const PriceContainer = styled.div`
    margin: ${({ theme }) => theme.spacing.get('md')} 0;
    margin-bottom: ${({ theme }) => theme.spacing.get('xl2')};

    @media screen and (max-width: ${breakpoints.get('lg')}) {
        margin-bottom: ${({ theme }) => theme.spacing.get('lg')};
    }
`;
