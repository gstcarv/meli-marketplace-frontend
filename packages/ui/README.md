# UI

Esse pacote contÃ©m os componentes base para a construÃ§Ãµes dos microfrontends.

## ğŸ“¦ InstalaÃ§Ã£o e ConfiguraÃ§Ã£o

### PrÃ©-requisitos

-   Node.js (versÃ£o 18.18.0+)
-   yarn

### Como rodar o storybook?

1. Instale as dependÃªncias:

    ```bash
    yarn install
    ```

2. Execute o storybook:

    ```bash
    yarn storybook
    ```

## ğŸ§ª Testes

Para rodar os testes, execute:

```bash
yarn test
```
