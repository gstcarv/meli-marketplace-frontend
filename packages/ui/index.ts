import './src/declarations/styled-components.d.ts';

export * from './src/components';

export * from './src/theme';
