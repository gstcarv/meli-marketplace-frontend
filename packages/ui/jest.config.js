module.exports = {
    testEnvironment: 'jsdom',
    transform: {
        '^.+\\.(j|t)sx?$': 'babel-jest'
    },
    moduleNameMapper: {
        '\\.(css)$': 'identity-obj-proxy',
        '\\.(png|jpg|jpeg|gif)$': 'identity-obj-proxy'
    },
    setupFilesAfterEnv: ['@testing-library/jest-dom', '<rootDir>/src/setupTests.ts']
};
