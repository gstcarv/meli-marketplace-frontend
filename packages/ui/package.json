{
    "name": "@meli/ui",
    "version": "0.0.0",
    "private": true,
    "main": "./index.ts",
    "scripts": {
        "lint": "eslint . --max-warnings 0",
        "test": "jest"
    },
    "devDependencies": {
        "@babel/preset-env": "^7.25.8",
        "@babel/preset-react": "^7.25.7",
        "@babel/preset-typescript": "^7.25.7",
        "@meli/design-tokens": "*",
        "@meli/eslint-config": "*",
        "@meli/icons": "*",
        "@meli/typescript-config": "*",
        "@testing-library/jest-dom": "^5.17.0",
        "@testing-library/react": "^12.1.2",
        "@turbo/gen": "^1.12.4",
        "@types/eslint": "^8.56.5",
        "@types/jest": "^27.0.1",
        "@types/node": "^20.11.24",
        "@types/react": "^18.2.61",
        "@types/react-dom": "^18.2.19",
        "babel-jest": "^29.7.0",
        "eslint": "^8.57.0",
        "jest": "^29.7.0",
        "jest-cli": "^29.7.0",
        "jest-environment-jsdom": "^29.7.0",
        "react-test-renderer": "^18.3.1",
        "typescript": "^5.3.3"
    },
    "peerDependencies": {
        "react": "^17.0.2",
        "styled-components": "^6.1.13"
    }
}
