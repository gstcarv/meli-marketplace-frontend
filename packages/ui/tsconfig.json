{
    "extends": "@meli/typescript-config/react-library.json",
    "compilerOptions": {
        "outDir": "dist",
        "module": "ES6",
        "moduleResolution": "Bundler"
    },
    "include": ["src", "./index.ts", "../design-tokens/src/lib"],
    "exclude": ["node_modules", "dist"]
}
