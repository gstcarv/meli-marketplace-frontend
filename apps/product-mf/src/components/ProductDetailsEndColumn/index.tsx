import { ProductItem } from '@marketplace/bff';
import { Button } from '@meli/ui';
import { useMemo } from 'react';
import { soldFormatter } from '../../helpers/soldFormatter';
import { PriceText } from '../PriceText';
import { Container, InfoHeading, InfoHeadingText, PriceContainer, ProductTitle } from './styled';

type Props = {
    product: ProductItem;
};

const ProductDetailsEndColumn = ({ product }: Props) => {
    const conditionLabel = useMemo(() => {
        if (product.condition === 'new') {
            return 'Novo';
        } else if (product.condition === 'used') {
            return 'Usado';
        }
    }, [product.condition]);

    const soldLabel = useMemo(() => soldFormatter(product.sold_quantity), [product.sold_quantity]);

    const headingLabel = useMemo(
        () => [conditionLabel, soldLabel].filter((item) => !!item).join(' | '),
        [conditionLabel, soldLabel]
    );

    return (
        <Container className='ProductDetails__price'>
            {headingLabel && (
                <InfoHeading data-testid='info-heading'>
                    <InfoHeadingText>{headingLabel}</InfoHeadingText>
                </InfoHeading>
            )}

            <ProductTitle>{product.title}</ProductTitle>

            <PriceContainer>
                <PriceText price={product.price} />
            </PriceContainer>

            <Button style={{ width: '100%' }}>Comprar</Button>
        </Container>
    );
};

export default ProductDetailsEndColumn;
