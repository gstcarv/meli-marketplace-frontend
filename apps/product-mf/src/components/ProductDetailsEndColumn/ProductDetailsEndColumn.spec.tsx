import { ThemeProvider } from '@meli/ui';
import { render, screen } from '@testing-library/react';
import ProductDetailsEndColumn from '.';
import { ProductItem } from '@marketplace/bff';

const productStub: ProductItem = {
    id: 'mock-id',
    condition: 'new',
    title: 'Mocked product title',
    price: {
        amount: 4445,
        currency: 'ARS'
    },
    free_shipping: true,
    picture: 'http://mock/picture.png',
    description: 'mock-description',
    sold_quantity: 100
};

describe('<ProductDetailsEndColumn />', () => {
    it('should render correctly with condition and sold quantity', () => {
        const { container } = render(
            <ThemeProvider>
                <ProductDetailsEndColumn product={{ ...productStub, sold_quantity: 30000, condition: 'new' }} />
            </ThemeProvider>
        );

        expect(container).toMatchSnapshot();
    });

    it('should render correctly when there is a big sold quantity', () => {
        render(
            <ThemeProvider>
                <ProductDetailsEndColumn product={{ ...productStub, condition: 'new', sold_quantity: 30000 }} />
            </ThemeProvider>
        );

        expect(screen.getByText('Novo | +30mil vendidos')).toBeInTheDocument();
    });

    it('should render correctly when there is a small sold quantity', () => {
        render(
            <ThemeProvider>
                <ProductDetailsEndColumn product={{ ...productStub, condition: 'used', sold_quantity: 16 }} />
            </ThemeProvider>
        );

        expect(screen.getByText(/Usado \| 16 vendidos/i)).toBeInTheDocument();
    });

    it('should render correctly when there is not condition either sold quantity provided', () => {
        render(
            <ThemeProvider>
                <ProductDetailsEndColumn
                    product={{
                        ...productStub,
                        sold_quantity: undefined,
                        condition: 'not_specified'
                    }}
                />
            </ThemeProvider>
        );

        expect(screen.queryByTestId('info-heading')).not.toBeInTheDocument();
    });
});
