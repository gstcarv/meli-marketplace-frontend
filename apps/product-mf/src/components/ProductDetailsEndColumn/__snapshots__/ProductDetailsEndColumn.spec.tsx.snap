// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDetailsEndColumn /> should render correctly with condition and sold quantity 1`] = `
.c8 {
  height: 48px;
  border: none;
  padding: 0 2.75rem;
  gap: 0 0.75rem;
  border-radius: 0.5rem;
  background-color: #4A7FEC;
  transition: background-color 0.8s;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
}

.c8:hover {
  background-color: #3a65ba;
}

.c9 {
  font-size: 1rem;
  font-weight: 500;
  color: #ffffff;
  letter-spacing: 0.3px;
}

.c5 {
  display: flex;
}

.c6 {
  color: #000000;
  font-size: 2.125rem;
  font-weight: 300;
}

.c7 {
  color: #000000;
  font-size: 0.875rem;
  font-weight: 300;
  margin-top: 0.125rem;
  margin-left: 0.125rem;
}

.c0 {
  width: 100%;
  flex: 0.25;
  padding: 1.5rem;
}

.c1 {
  display: flex;
}

.c2 {
  display: flex;
  color: #9A9897;
  font-size: 0.875rem;
  font-weight: 300;
  margin-bottom: 0.5rem;
}

.c3 {
  color: #000000;
  font-size: 1.375rem;
  font-weight: 500;
  margin-bottom: 0;
}

.c4 {
  margin: 1rem 0;
  margin-bottom: 2.75rem;
}

<div>
  <section
    class="c0"
  >
    <div
      class="c1"
      data-testid="info-heading"
    >
      <span
        class="c2"
      >
        Novo | +30mil vendidos
      </span>
    </div>
    <h1
      class="c3"
    >
      Mocked product title
    </h1>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <span
          class="c6"
        >
          $ 4.445
        </span>
        <span
          class="c7"
        >
          00
        </span>
      </div>
    </div>
    <button
      class="c8"
      style="width: 100%;"
    >
      <span
        class="c9"
      >
        Comprar
      </span>
    </button>
  </section>
</div>
`;
